homeassistant/components/climate/zhong_hong.py

```diff
@@ -0,0 +1,179 @@
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, startup)
+
+
+class ZhongHongClimate(ClimateDevice):
+    """Representation of a ZhongHong controller support HVAC."""
+
+    def __init__(self, hub, addr_out, addr_in):
+        """Set up the ZhongHong climate devices."""
+        from zhong_hong_hvac.hvac import HVAC
+        self._device = HVAC(hub, addr_out, addr_in)
+        self._hub = hub
+        self._device.register_update_callback(self._after_update)
+
+    def _after_update(self, climate):
+        """Callback to update state."""
+        _LOGGER.debug("async update ha state")
+        self.schedule_update_ha_state()
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self.unique_id
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the HVAC."""
+        return "zhong_hong_hvac_{}_{}".format(self._device.addr_out,
+                                              self._device.addr_in)
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return (SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+                | SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._device.current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._device.operation_list
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._device.current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._device.target_temperature
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self._device.is_on
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return self._device.current_fan_mode
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return self._device.fan_list
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return convert_temperature(self._device.min_temp, TEMP_CELSIUS,
+                                   self.temperature_unit)
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return convert_temperature(self._device.max_temp, TEMP_CELSIUS,
```

Comment:
You don't need to convert anything.  Your device is already in CELSIUS and you are converting to CELSIUS. Just return `self._device_max_temp`

Reply:
fixed